### PR TITLE
adding in go build yaml and necessary make cmds

### PR DESCRIPTION
**Pull request by acornett21**
- adding a proper go build process and make file changes for `lint`, `fmt`, `tidy`, and `cover`
- fixing any/all issues found from any of the cmd's in the build process

### CODE REVIEW

File: cmd/decrypt.go [hunk 1]
```diff
@@ -4,11 +4,12 @@ Copyright © 2022 NAME HERE <EMAIL ADDRESS>
...
 
 import (
-	"github.com/ProtonMail/gopenpgp/v2/crypto"
-	"github.com/gobuffalo/envy"
 	"log"
 	"os"
 
+	"github.com/ProtonMail/gopenpgp/v2/crypto"
+	"github.com/gobuffalo/envy"
+
```
Comment (bcrochet) [inline cmd/decrypt.go:12]: This empty line can also go away.

File: cmd/decrypt.go [hunk 2]
```diff
@@ -63,12 +64,24 @@ func decryptRun(cmd *cobra.Command, args []string) {
 	var publickeyobj *crypto.Key
 	var publickeyring *crypto.KeyRing
 	publickeyobj, err = crypto.NewKeyFromArmored(string(decryptionpublickey))
+	if err != nil {
```
Comment (bcrochet) [inline cmd/decrypt.go:67]: Not necessary here, but I feel like this needs to become a RunE and we return the errors, rather than Fatal.

**Review by bcrochet — COMMENTED** [review]
No blocking changes requested. Feel free to mark as resolved, or address them.

File: cmd/decrypt.go (hunk 2 above)
Comment (acornett21) [reply cmd/decrypt.go:66]: This makes more sense to me as well, then we can have cobra worry about processing the error. I can create an issue for this.

- Relates: #16

File: cmd/artifacts.go [hunk 3]
```diff
@@ -7,14 +7,16 @@ import (
...
 	"io"
-	pjapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
 	"log"
 	"net/http"
 	"os"
 	"path/filepath"
 	"strings"
+
+	"github.com/spf13/cobra"
+
```
Comment (bcrochet) [inline cmd/artifacts.go:18]: These should not be separated.

File: cmd/root.go [hunk 4]
```diff
@@ -4,11 +4,12 @@ Copyright © 2022 NAME HERE <EMAIL ADDRESS>
...
 
 import (
-	configflagutil "k8s.io/test-infra/prow/flagutil/config"
 	"log"
 	"os"
 	"reflect"
 
+	configflagutil "k8s.io/test-infra/prow/flagutil/config"
+
 	"github.com/spf13/cobra"
```
Comment (bcrochet) [inline cmd/root.go:13]:
```suggestion
	configflagutil "k8s.io/test-infra/prow/flagutil/config"
	"github.com/spf13/cobra"
```

**Review by bcrochet — CHANGES_REQUESTED** [review]
Couple of comments.

File: cmd/artifacts.go (hunk 3 above)
Comment (acornett21) [reply cmd/artifacts.go]: resolved.

File: cmd/root.go (hunk 4 above)
Comment (acornett21) [reply cmd/root.go]: resolved.

**Review by bcrochet — APPROVED** [review]
/lgtm

**Pull request merged**